File: exercises/D403/solution.c
#include <stdio.h>
#include <string.h>
int freqa[26], freqb[26];
char s[11];
int main() {
    scanf("%s", s);
    int len = strlen(s);
    int n, i, j;
    for (i = 0; i < len; i++) {
        freqa[s[i] - 'A']++;
    }
    scanf("%d", &n);
    for (i = 0; i < n; i++) {
        for (j = 0; j < 26; j++) {
            freqb[j] = 0;
        }
        scanf("%s", s);
        len = strlen(s);
        for (j = 0; j < len; j++) {
            freqb[s[j] - 'A']++;
        }
        int ok = 1;
        for (j = 0; j < 26; j++) {
            if (freqb[j] > freqa[j]) {
                ok = 0;
            }
        }
        if (ok) {
            printf("Yes\n");
        } else {
            printf("No\n");
        }
    }
    return 0;
}

File: exercises/D403/solution.java
import java.util.*;

class D403 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s = in.nextLine();
        int[] freqa = new int[26];
        int[] freqb = new int[26];
        int len = s.length();
        for (int i = 0; i < len; i++) {
            freqa[s.charAt(i) - 'A']++;
        }
        int n = in.nextInt();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < 26; j++) {
                freqb[j] = 0;
            }
            s = in.next();
            len = s.length();
            for (int j = 0; j < len; j++) {
                freqb[s.charAt(j) - 'A']++;
            }
            boolean ok = true;
            for (int j = 0; j < 26; j++) {
                if (freqb[j] > freqa[j]) {
                    ok = false;
                }
            }
            if (ok) {
                System.out.println("Yes");
            } else {
                System.out.println("No");
            }
        }
    }
}
